gobblin-runtime/src/main/java/gobblin/scheduler/JobScheduler.java

```diff
@@ -377,11 +373,9 @@ public void runJob(Properties jobProps, JobListener jobListener, JobLauncher job
       return;
     }
 
-    // Populate the assigned job ID
-    jobProps.setProperty(ConfigurationKeys.JOB_ID_KEY, JobLauncherUtils.newJobId(jobName));
-
     // Launch the job
     try (Closer closer = Closer.create()) {
+      closer.register(MDC.putCloseable(ConfigurationKeys.JOB_NAME_KEY, jobName));
```

Comment:
Why separate the two? lauchJob() is called in the next line. When lauchJob() is called in a different context then won't you miss the job name unless you push the job name in launchJob() or the callers of launchJob()?